design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,527 @@
+    text: String @doc(description: "ACL resource label.")
+    sortOrder: Int @doc(description: "ACL resource sort order.")
+    children: [CompanyAclResource!] @doc(description: "An array of sub-resources.")
+}
+
+type CompanyUserEmailCheckResponse @doc(description: "Response object schema for a Company User email validation query.") {
+    status: Int @doc(description: "1 - 'OK', 0 - 'Error'")
+    message: String @doc(description: "Error or notice message for a User.")
+}
+
+type CompanyHierarchyOutput @doc(description: "Response object schema for a Company Hierarchy query.") {
+    structure: CompanyHierarchyElement @doc(description: "An array of Company structure elements.")
+    draggable: Boolean @doc(description: "Flag that defines whether Company Hierarchy can be changed by current User or not.")
+    max_nesting: Int @doc(description: "Indicator of maximun nesting of elements within a whole Company Hierarchy.")
+}
+
+type CompanyHierarchyElement @doc(description: "Company Hierarchy element output data schema.") {
+    id: Int @doc(description: "Hierarchy element ID.")
+    tree_id: Int @doc(description: "The hierarchical ID of the element within a structure. Used for changing element's position in hierarchy.")
+    type: String @doc(description: "Hierarchy element type: 'customer' or a 'team'.")
+    text: String @doc(description: "Hierarchy element name.")
+    description: String @doc(description: "Hierarchy element description.")
+    opened: Boolean @doc(description: "Flag that defines whether Hierarchy element should be expanded or not.")
+    children: [CompanyHierarchyElement!] @doc(description: "An array of child elements.")
+}
+
+type CompanyTeam @doc(description: "Company Team entity output data schema.") {
+    id: Int @doc(description: "Team ID.")
+    name: String @doc(description: "Team name.")
+    description: String @doc(description: "Team description.")
+}
+
+type SearchResultPageInfo @doc(description: "Output data schema for a meta data object of the search query response.") {
+    page_size: Int @doc(description: "Maximum number of items returned by a search query.")
+    current_page: Int @doc(description: "Current result page number.")
+    total_pages: Int @doc(description: "Total number of pages.")
+}
+
+input CompanyUsersFilterInput @doc(description: "Defines the input filters for a Company Users query") {
+    status: CompanyUserStatusFilterValuesEnum! @doc(description: "Filter Customers by their status within a Company structure. Defaults to 'active'. Required.")
+}
+
+enum CompanyUserStatusFilterValuesEnum @doc(description: "List of available Company user statuses.") {
+    active @doc(description: "Only active users")
+    inactive @doc(description: "Only inactive users")
+    all @doc(description: "All users")
+}
+```
+
+# Mutations
+
+```graphql
+type Mutation {
+    createCompany(input: CompanyCreateInput!): Company  @doc(description:"Create new Company.")
```

Comment:
Per https://github.com/magento/architecture/blob/master/design-documents/graph-ql/extensibility.md, arguments should not be wrapped into wrapper objects unless it makes sense from business perspective. 
Outputs of mutations MUST be wrapped for extensibility purposes. Please correct for all mutations here.

Also, even if we preserve some inputs, the convention is to name the type using `mutation name` + `Input` suffix. Currently words in mutation name are swapped.